sympy/functions/combinatorial/numbers.py

```diff
@@ -1984,3 +1984,123 @@ def nT(n, k=None):
     for discard in m.enum_range(n[_M], k-1, k):
         tot += 1
     return tot
+
+#-----------------------------------------------------------------------------#
+#                                                                             #
+#                          Motzkin numbers                                    #
+#                                                                             #
+#-----------------------------------------------------------------------------#
+class motzkin(Function):
+    """
+    In mathematics, the nth Motzkin number is the number
+    of different ways of drawing non-intersecting chords
+    between n points on a circle (not necessarily touching
+    every point by a chord). The Motzkin numbers are named
+    after Theodore Motzkin and have diverse applications
+    in geometry, combinatorics and number theory.
+
+    Motzkin numbers satisfy this recurrence relation
+    M(n) = ( (2n+1) / (n+2) )*M(n-1)  +  ( (3n-3) / (n+2) )*M(n-2)
+
+
+    Examples
+    ========
+
+    >>> from sympy import motzkin
+    >>> motzkin.find_first_n_motzkin(10)
```

Comment:
Maybe we could make these play better with itertools (or similar in SymPy). Have an iteratator that yields these numbers in order, `m = motzkin.numbers(); take(m, 5) -> [1, 1, 2, 4, 9]`.

For the find functionality, we can drop the word range: `motzkin.range` like `primerange` has the normal Python behavior. I would recommend making the arg polymorphic so `motzkin.range(3)` behaves like `range(3)`.